multicluster/cmd/check.go

```diff
@@ -464,12 +466,24 @@ func (hc *healthChecker) checkIfGatewayMirrorsHaveEndpoints(ctx context.Context)
 			errors = append(errors, fmt.Errorf("%s.%s mirrored from cluster [%s] has no endpoints", svc.Name, svc.Namespace, svc.Labels[k8s.RemoteClusterNameLabel]))
 			continue
 		}
+
+		vizNs, err := hc.linkerdHC.KubeAPIClient().GetNamespaceWithExtensionLabel(ctx, vizCmd.ExtensionName)
+		if err != nil {
+			errors = append(errors, fmt.Errorf("failed to fetch gateway metrics: %s", err))
```

Comment:
I don't have a working multicluster setup right now so I can't test this, but just reading this it looks like it would return an error and fail the check if the viz extension is not installed.  I'd expect that if the viz extension is not installed that this would return a `healthcheck.SkipError` to skip this check.

Reply:
Yeah, currently it would show an error as shown [above](https://github.com/linkerd/linkerd2/pull/5560#issuecomment-764206298). We could also make it a warning and still have the overall `linkerd mc check` succeed, which might be more explicit. WDYT?

Reply:
I would return `SkipError`.  There's no hard dependency on viz from multicluster so if viz is unavailable, that shouldn't be considered a problem.  It just means that if viz *is* available, then we can do some extra validation of the stats.